lib/modules/versioning/kubernetes-api/index.spec.ts

```diff
@@ -0,0 +1,137 @@
+import { KubernetesApiVersioningApi } from './index';
+
+describe('modules/versioning/kubernetes-api/index', () => {
+  const versioning = new KubernetesApiVersioningApi();
+
+  test.each`
+    version       | expected
+    ${'v1'}       | ${true}
+    ${'v2'}       | ${true}
+    ${'v1alpha1'} | ${false}
+    ${'v1beta11'} | ${false}
+  `('isStable("$version") === $expected', ({ version, expected }) => {
+    expect(versioning.isStable(version)).toBe(expected);
+  });
+
+  test.each`
+    version                          | expected
+    ${'v1'}                          | ${true}
+    ${'v2'}                          | ${true}
+    ${'v10'}                         | ${true}
+    ${'v1.0'}                        | ${false}
+    ${'1'}                           | ${false}
+    ${'2'}                           | ${false}
+    ${'10'}                          | ${false}
+    ${'1.0'}                         | ${false}
+    ${'apps/v1'}                     | ${true}
+    ${'telemetry.istio.io/v1alpha1'} | ${true}
+    ${'k3d.io/v1alpha2'}             | ${true}
+    ${'extensions/v1beta1'}          | ${true}
+    ${'apps/v1beta2'}                | ${true}
+    ${'autoscaling/v2'}              | ${true}
+    ${'acme.cert-manager.io/v1'}     | ${true}
+  `('isValid("$version") === $expected', ({ version, expected }) => {
+    expect(versioning.isValid(version)).toBe(expected);
+  });
+
+  test.each`
+    version       | major | minor | patch
+    ${'v1'}       | ${1}  | ${0}  | ${0}
+    ${'v2'}       | ${2}  | ${0}  | ${0}
+    ${'v1alpha1'} | ${1}  | ${0}  | ${0}
+    ${'v1alpha2'} | ${1}  | ${0}  | ${0}
+    ${'v1beta1'}  | ${1}  | ${0}  | ${0}
+    ${'v1beta2'}  | ${1}  | ${0}  | ${0}
+  `(
+    'getMajor, getMinor, getPatch for "$version"',
+    ({ version, major, minor, patch }) => {
+      expect(versioning.getMajor(version)).toBe(major);
+      expect(versioning.getMinor(version)).toBe(minor);
+      expect(versioning.getPatch(version)).toBe(patch);
+    }
+  );
+
+  test.each`
+    version       | other         | expected
+    ${'v1'}       | ${'v1'}       | ${true}
+    ${'v1'}       | ${'v2'}       | ${false}
+    ${'v1'}       | ${'v1alpha1'} | ${false}
+    ${'v1'}       | ${'v1alpha2'} | ${false}
+    ${'v1'}       | ${'v1beta1'}  | ${false}
+    ${'v1'}       | ${'v1beta2'}  | ${false}
+    ${'v1alpha1'} | ${'v1alpha1'} | ${true}
+    ${'v1alpha1'} | ${'v1alpha2'} | ${false}
+    ${'v1alpha1'} | ${'v1beta1'}  | ${false}
+    ${'v1alpha1'} | ${'v1beta2'}  | ${false}
+  `(
+    'equals("$version", "$other") === $expected',
```

Comment:
maybe this should have a test with compabillity prefix too?

Reply:
Added.

It feels a little wrong that `apps/v1 === autoscaling/v1`, but I guess that's what `isCompatible` is for 🤷

Reply:
ah, yes. they are equal but not compatible, as `isCompatible` would return false and renovate won't consider to update `apps/v1` to `autoscaling/v2`